### PR TITLE
[mono][jit] Adding Vector128.GetElement as intrinsic in arm64.

### PR DESCRIPTION
**Pull request by jandupej**
**EDITED**

Since arm64 has no instruction to select from SIMD lanes by a GP register, this does the following to achieve `Vector128.GetElement`:
- If the index is a constant and within range, elides bounds check and translates into `OP_EXTRACT_x` instead of `OP_XEXTRACT_x`. The result is a single instruction.
- If the index is a constant and out of range, does the same as the above, but includes a bounds check.
- If index is non-constant, but the result is 64-bit, this selects the upper or lower half of the register with a conditional branch. This yields 4 instructions or just 2 if extracting a `double` and `sreg1==dreg`.
- Otherwise spills the vector reg and reads back the correct element.

Contributes to https://github.com/dotnet/runtime/issues/80566

### CODE REVIEW

**Comment by tannergooding** [comment]
> If the element type is small, the switch can have as many as 16 cases, resulting in 140 Bytes emitted for GetElement<byte>. For float and int, we emit 44 Bytes. This can be optimized for size if needed.

In RyuJIT we specially handle `GetElement` for non-constants by emiting a bounds check, spilling, and loading the relevant offset from memory. It works rather nicely for the "rare" case and avoids us needing to emit a large jump table. Since we already had the optimization to efficiently handle the value already being in memory, this didn't require "more code" in practice.

File: src/mono/mono/mini/mini-arm64.c [hunk 1]
```diff
@@ -926,6 +927,94 @@ mono_arm_emit_ldrx (guint8 *code, int rt, int rn, int imm)
 	return emit_ldrx (code, rt, rn, imm);
 }
 
+static guint8* 
+emit_switch (guint8* code, int ncases, int ncaselen, int sreg)
+{
```
Comment (vargaz) [inline src/mono/mono/mini/mini-arm64.c:932]: This is now unused.

Comment (jandupej) [reply src/mono/mono/mini/mini-arm64.c]: gone

**Review by vargaz — APPROVED** [review]

**Comment by jandupej** [comment]
CI failures are unrelated.

**Pull request merged**